lib/docker/imageservice.go

```diff
@@ -62,6 +66,36 @@ type RegistryConnectionRequest struct {
 	ClientCertPath string
 	// ClientKeyPath is the full path to the client private key
 	ClientKeyPath string
+	// Username specifies optional registry username for basic auth
+	Username string
+	// Password specifies optional registry password for basic auth
+	Password string
+	// Domain specifies optional registry prefix when pushing images
+	Domain string
+	// Insecure indicates a plain http registry
+	Insecure bool
+}
+
+// HasBasicAuth returns true if the request contains basic auth credentials.
+func (r *RegistryConnectionRequest) HasBasicAuth() bool {
+	return r.Username != "" && r.Password != ""
+}
+
+// TLSClientConfig returns client TLS config for this request.
+func (r *RegistryConnectionRequest) TLSClientConfig() (*tls.Config, error) {
+	certificate, err := tls.LoadX509KeyPair(r.ClientCertPath, r.ClientKeyPath)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	roots, err := newCertPool([]string{r.CACertPath})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &tls.Config{
+		Certificates: []tls.Certificate{certificate},
+		MinVersion:   tls.VersionTLS10,
```

Comment:
Min version should be TLS 1.2 or 1.3. TLS 1.3 is modern compatability, which is what we should be using if we fully control the system. Otherwise 1.2 is acceptable for intermediate compatability.

For TLS recommended settings we've been following https://wiki.mozilla.org/Security/Server_Side_TLS and also need to hard code the settings (golang doesn't necessarily choose sensible defaults from what I recall).